armi/physics/neutronics/energyGroups.py

```diff
@@ -275,14 +275,391 @@ def getGroupStructureType(neutronEnergyBoundsInEv):
 )
 
 # fmt: on
+def _create_multigroup_structures_on_finegroup_energies(
+    multigroup_energy_bounds, finegroup_energy_bounds
+):
+    """Set energy group bounds to the nearest ultra-fine group boundaries."""
+    modifiedEnergyBounds = set()
+    modifiedEnergyBounds.add(max(finegroup_energy_bounds))
+    for energyBound in multigroup_energy_bounds[1:]:
+        modifiedEnergyBounds.add(
+            utils.findNearestValue(finegroup_energy_bounds, energyBound)
+        )
+
+    return sorted(modifiedEnergyBounds, reverse=True)
+
+
 def _create_anl_energies_with_group_energies(group_energy_bounds):
     """Set energy group bounds to the nearest ultra-fine group boundaries."""
     ufgEnergies = _create_anl_energies_with_group_lethargies(itertools.repeat(1, 2082))
-    modifiedEnergyBounds = []
-    for energyBound in group_energy_bounds:
-        modifiedEnergyBounds.append(utils.findNearestValue(ufgEnergies, energyBound))
-    return modifiedEnergyBounds
+    return _create_multigroup_structures_on_finegroup_energies(
+        group_energy_bounds, ufgEnergies
+    )
+
 
+GROUP_STRUCTURE["SHEM361"] = [
```

Comment:
Can you just give a reference for SHEM361 in a comment here?

Reply:
@ntouran -- that's a good idea (its mildly tricky to find for free); here you go.